typescript/cli/src/context/altvm.ts

```diff
@@ -192,49 +165,45 @@ export class AltVMSignerFactory
     strategyConfig: Partial<ExtendedChainSubmissionStrategy>,
   ) {
     const signers: ChainMap<AltVM.ISigner<AnnotatedTx, TxReceipt>> = {};
-
     for (const chain of chains) {
       const metadata = metadataManager.getChainMetadata(chain);
 
-      const protocol = ALT_VM_SUPPORTED_PROTOCOLS[metadata.protocol];
-
-      if (!protocol) {
+      if (!hasProtocol(metadata.protocol)) {
         continue;
       }
 
-      const privateKey = await AltVMSignerFactory.loadPrivateKey(
-        keyByProtocol,
-        strategyConfig,
-        metadata.protocol,
-        chain,
-      );
+      const signerConfig = {
+        privateKey: await AltVMSignerFactory.loadPrivateKey(
+          keyByProtocol,
+          strategyConfig,
+          metadata.protocol,
+          chain,
+        ),
+      };
 
-      signers[chain] = await protocol.signer.connectWithSigner(
-        metadata.rpcUrls.map((rpc) => rpc.http),
-        privateKey,
-        {
-          metadata,
-        },
-      );
+      signers[metadata.protocol] = await getProtocolProvider(
```

Comment:
We still need signers created per chain because they use chain metadata to get RPC URLs, etc.